dash/backend/src/modules/auth/services/oauth2/google-oauth2.service.ts

```diff
@@ -43,9 +44,15 @@ export class GoogleOauth2Service extends Oauth2AuthProvider {
 
     userProfile.authorities = [];
 
+    // Check access rights for profile
     const userAuthority = new UserAuthority();
-    userAuthority.id = 3;
-    userProfile.authorities.push(userAuthority);
+    const emailDomain = userProfile.email.split('@').pop().toLowerCase().trim();
+    if (oAuth2Config.allowedDomains.includes(emailDomain)) {
```

Comment:
may want to make sure the allowed domains are lowercased, too, if its not doin gthat already

Reply:
The allowed domains are validated to be lowercased in the external auth config dto

```
@IsNotEmpty()
@IsString({each: true})
@IsLowercase({each: true})
allowedDomains: string[];
```

And when creating a config, the domains are changed to lowercase before passing it to the backend. Is there anywhere else they should be validated for that?

Reply:
That works!